pkg/reconciler/ingress/ingress.go

```diff
@@ -0,0 +1,279 @@
+				if errors.IsAlreadyExists(err) {
+					existingLeaf, err := c.client.NetworkingV1().Ingresses(desiredleaf.Namespace).Get(ctx, desiredleaf.Name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+
+					// Set the resourceVersion and UID to update the desired leaf.
+					desiredleaf.ResourceVersion = existingLeaf.ResourceVersion
+					desiredleaf.UID = existingLeaf.UID
+
+					if _, err := c.client.NetworkingV1().Ingresses(desiredleaf.Namespace).Update(ctx, desiredleaf, metav1.UpdateOptions{}); err != nil {
+						return err
+					}
+
+				} else {
+					return err
+				}
+			}
+		}
+
+	} else {
+		// If the ingress has the clusterLabel set, that means that it is a leaf and it's synced with
+		// a cluster.
+		//
+		// This update can come from the creation or because the syncer has update the status.
+
+		rootIngressName := ingress.Labels[ownedByLabel]
+		// A leaf Ingress was updated; get others and aggregate status.
+		sel, err := labels.Parse(fmt.Sprintf("%s=%s", ownedByLabel, rootIngressName))
+		if err != nil {
+			return err
+		}
+		others, err := c.lister.List(sel)
+		if err != nil {
+			return err
+		}
+
+		// Get the rootIngress based on the labels.
+		var rootIngress *networkingv1.Ingress
+
+		rootIf, exists, err := c.indexer.Get(&networkingv1.Ingress{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:   ingress.Namespace,
+				Name:        rootIngressName,
+				ClusterName: ingress.GetClusterName(),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// TODO(jmprusi): A leaf without rootIngress?
+		if !exists {
+			return fmt.Errorf("Root Ingress not found: %s", rootIngressName)
+		}
+
+		rootIngress = rootIf.(*networkingv1.Ingress).DeepCopy()
+
+		// Clean the current status, and then recreate if from the other leafs.
+		rootIngress.Status.LoadBalancer.Ingress = []corev1.LoadBalancerIngress{}
+		for _, o := range others {
+			rootIngress.Status.LoadBalancer.Ingress = append(rootIngress.Status.LoadBalancer.Ingress, o.Status.LoadBalancer.Ingress...)
+		}
+
+		// If the envoy controlplane is enabled, we update the cache and generate and send to envoy a new snapshot.
+		if c.envoycontrolplane != nil {
+			err := c.envoycontrolplane.UpdateIngress(*rootIngress)
+			if err != nil {
+				return err
+			}
+
+			statusHost := generateStatusHost(c.domain, rootIngress)
+			// Now overwrite the Status of the rootIngress with our desired LB
+			rootIngress.Status.LoadBalancer.Ingress = []corev1.LoadBalancerIngress{{
+				Hostname: statusHost,
+			}}
+		}
+
+		// Update the rootIngress status with our desired LB.
+		if _, err := c.client.NetworkingV1().Ingresses(rootIngress.Namespace).UpdateStatus(ctx, rootIngress, metav1.UpdateOptions{}); err != nil {
+			if errors.IsConflict(err) {
+				key, err := cache.MetaNamespaceKeyFunc(ingress)
+				if err != nil {
+					return err
+				}
+				c.queue.AddRateLimited(key)
+				return nil
+			}
+			return err
```

Comment:
You should just be able to `return err` here - in both flows, the key will be added with rate limiting.